internal/runner/nomad_runner.go

```diff
@@ -130,6 +130,7 @@ func (r *NomadJob) UpdateFileSystem(copyRequest *dto.UpdateFileSystemRequest) er
 	stdOut := bytes.Buffer{}
 	stdErr := bytes.Buffer{}
 	exitCode, err := r.api.ExecuteCommand(r.id, context.Background(), updateFileCommand, false,
+		nomad.PrivilegedExecution, // All files should be written and owned by a privileged user #211.
```

Comment:
Is this not an oversimplification? While test files are not supposed to be writable (and readable?), one can well imagine a task in which a given CSV file is to be extended by entries... This distinction is currently not possible. @MrSerth Do we want to extend our API for the UpdateFileRoute by an Owner (, Group and Permissions)?

Reply:
Hm, good point. 

- Test files need to be readable by `user`, as this user is supposed to execute them. Hence, we cannot remove that flag.
- With regard to the example you provided: Yes, the current implementation is a simplification. However, I am currently not aware of a use case where appending is required, but this might be valid. Instead of adding an explicit owner, group and permissions, I would go a soft way: What if we add one additional boolean flag (besides `path` and `content`) to the UpdateFileRoute indicating whether the user used for execution should have write permissions? Or which permissions (`rwx`) the user should haven general? I would just prevent providing an explicit user name, as this requires further configuration. If set, this could be apply to the "other" group and will therefore work for the `user`. If privilegedExecution is enabled, the "other" group won't be respected, but `root` may always read :)